webapp/controller/PageOverview.controller.js

```diff
@@ -0,0 +1,199 @@
+		},
+
+		getCombinedFilter: function(aSelectedId) {
+			const sQuerySearch = this.byId("searcher").getValue();
+			const sSelectedKey = this.byId("availabilitySelect").getSelectedKey();
+			
+			const sAvailability = "all";
+			const oAvailabilityFilter = new Filter(
+				"Availability", 
+				FilterOperator.NE,
+				sSelectedKey === sAvailability ? sSelectedKey : JSON.parse(sSelectedKey.toLowerCase())
+			)
+			const aSearchFilters = [
+				new Filter("Name", FilterOperator.Contains, sQuerySearch),
+				new Filter("Descrition", FilterOperator.Contains, sQuerySearch)
+			]
+			const oSearchFilter = new Filter({
+				filters: aSearchFilters,
+				and: false
+			});
+			const oCategoryFilters = new Filter({
+				filters: this.getCategoriesFilters(),
+				and: false
+			})
+			
+			const oSuppliersFilters = new Filter({
+				filters: this.getSuppliersFilters(aSelectedId),
+				and: true
+			})
+			return new Filter({
+				filters: [oSearchFilter, oAvailabilityFilter, oCategoryFilters, oSuppliersFilters],
+				and: true
+			});
+		},
+
+		onFilter: function(aSelectedId = null) {
+			const oTable = this.byId("table");
+			const oItemsBinding = oTable.getBinding("items");
+			
+			const oFilter = this.getCombinedFilter(aSelectedId);
+			
+			oItemsBinding.filter(oFilter);
+		},
+
+		getSuppliersFilters: function(aSelectedId) {
+			const oMultiInput = this.byId("multiInput");
+			const oModel = this.getView().getModel();
+			const aAllSuppliers = oModel.getProperty("/Suppliers");
+			const aSuppliersItems = oMultiInput.getTokens().map(el => el.getText());
+			const aFilters = [];
+			let aSuppliersId = null;
+		
+			if(aSelectedId) {
+				aSuppliersId = aSelectedId;
+			} else {
+				aSuppliersId = aAllSuppliers
+										.filter(el => aSuppliersItems.includes(el.SuppliersName))
+										.map(el => el.SupplierId);
+										
+			}
+		
+			
+			
+			aSuppliersId.forEach((el) => {
+				aFilters.push(
+					new Filter({
+						path: "Supplier",
+						operator: FilterOperator.EQ,
+						value1: el,
+						test: (supplier) => {
+							const aResult = supplier.filter((item) => {
+							
+								return	item === el
+							});
+
+							return !!aResult.length;
+						}
+					})
+				)
+			})
+
+			return aFilters.length ? aFilters : [new Filter("Supplier/0", FilterOperator.Contains, "")];
+		},
+
+		getCategoriesFilters: function() {
+			const oCategorySelect = this.byId("categorySelect");
+			const aCategoryItems = oCategorySelect.getSelectedItems();
+			
+			const aFilters = aCategoryItems.map((el) => {
+				return new Filter("Category", FilterOperator.Contains, el.getText())
+			})
+			
+			return aFilters.length ? aFilters : [new Filter("Category", FilterOperator.Contains, "")];
+		},
+
+		
+
+		handleValueHelp: function(event) {
+			const oView = this.getView();
+			const oModel = oView.getModel();
+			if (!this.oDialog) {	
+				this.oDialog = sap.ui.xmlfragment(oView.getId(), "products.app.view.fragments.SuppliersDialog", this);
```

Comment:
do not use sap.ui.xmlfragment as it's deprecated. Use Fragment.load instead. Find api in the docs